tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -351,6 +364,90 @@ find_attr_pos_in_tlist(List *targetlist, AttrNumber pos)
 	pg_unreachable();
 }
 
+static bool
+qual_is_vectorizable(DecompressChunkPath *path, Node *qual)
+{
+	/* Only simple "Var op Const" binary predicates for now. */
+	if (!IsA(qual, OpExpr))
+	{
+		return false;
+	}
+
+	OpExpr *o = castNode(OpExpr, qual);
+
+	if (list_length(o->args) != 2)
```

Comment:
Could we have a test case for this case?

Reply:
Added a case with unary operator.